dali/operators/util/axis_args.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_UTIL_AXIS_ARGS_H_
+#define DALI_OPERATORS_UTIL_AXIS_ARGS_H_
+
+#include <memory>
+#include <vector>
+
+#include "dali/core/small_vector.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+class AxisArgs {
+ public:
+  enum Flags : unsigned {
+    AllowEmpty = 0b0001,
+    AllowMultiple = 0b0010,
+    AllIfEmpty = 0b0100,
+    AllowNegative = 0b1000,
+    DefaultFlags = AllowEmpty | AllowMultiple | AllIfEmpty | AllowNegative
+  };
+
+  AxisArgs(const OpSpec &spec, const char *axes_arg, const char *axis_names_arg,
```

Comment:
Nitpick:
I suggested some plurality-neutral names (like `index_arg` and `name_arg`) because, in principle, this helper can be used to get just one axis.